Add architecture diagram into README

r-ash opened the pull request.

M-Kusumgar reviewed (commented):
Looks good as it is happy to approve, couple of small things that are optional to address are:
* putting hintr label on the R API
* rrq label on Queue
* hintr-worker-XXXX on worker pool just so people know what the containers are called
* mention of vuejs frontend

r-ash commented:
Thanks, updated that now. I've added a hintr label on the API and workers, added the front end. Left off rrq though. That is just a library used by the hintr API as that is just a library that we use instead of a layer in the architecture
